yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -656,23 +661,40 @@ func (yb *TargetYugabyteDB) importBatchFastRecover(conn *pgx.Conn, batch Batch,
 			}
 		}
 
-		fields := strings.Split(line, args.Delimiter)
-		stmtArgs := make([]interface{}, len(fields))
-		for i, val := range fields {
-			stmtArgs[i] = val
+		var singleLineReader io.Reader
+		if copyHeader != "" {
+			singleLineReader = strings.NewReader(copyHeader + "\n" + line)
+		} else {
+			singleLineReader = strings.NewReader(line)
 		}
 
-		// 5. Execute the INSERT statement one by one
-		res, err := conn.Exec(context.Background(), insertStmt, stmtArgs...)
+		// 5. Execute the COPY statement with the line read from the file
+		res, err := conn.PgConn().CopyFrom(context.Background(), singleLineReader, copyCommand)
 		if err != nil {
-			return 0, fmt.Errorf("insert into target table: %w", err)
+			// Ignore err if VIOLATES_UNIQUE_CONSTRAINT_ERROR only
+			if strings.Contains(err.Error(), VIOLATES_UNIQUE_CONSTRAINT_ERROR) {
+				// logging lineNum might not be useful as batches are truncated later on
+				log.Debugf("ignoring error %q for line=%q in batch %q", err.Error(), line, batch.GetFilePath())
+				rowsIgnored++ // increment before continuing to next line
+				continue
+			}
+
+			var pgerr *pgconn.PgError
+			if errors.As(err, &pgerr) {
+				err = fmt.Errorf("%s, %s in %s", err.Error(), pgerr.Where, batch.GetFilePath())
+			}
+			return rowsAffected + rowsIgnored, err
 		}
 
+		// rowsAffected will be either 0 or 1
 		if res.RowsAffected() > 0 {
 			rowsAffected++
 		} else {
+			// at this point it is not expected to have 0 rows affected
```

Comment:
should we error out here?

Reply:
hmm, since it is unknown and unexpected case?
Can error policy take care of this?

Reply:
on second thoughts, it's probably okay to warn. We also do that in `logDiscrepancyInEventBatchIfAny` . It is definitely unexpected. COPY should not return 0 rows affected. 

Error policy will kick in anyways if there is an error(at the FileTaskImporter.importBatch layer). I didn't understand what you mean by "take care of this"?